Src/Sarif.PatternMatcher.Sdk/DynamicValidatorBase.cs

```diff
@@ -16,6 +16,7 @@ public abstract class DynamicValidatorBase : StaticValidatorBase
             resultLevelKind = default;
 
             if (shouldUseDynamicCache &&
+                FingerprintToResultCache != null &&
```

Comment:
>FingerprintToResultCache [](http://example.com/codeflow?start=16&length=24)

Is this change needed?

looking at the changes in the ValidatorBase, you still initialize the FingerprintToResultCache in the constructor.